### PR TITLE
Split common columns documentation

### PR DESCRIPTION
**Pull request by raamSoftwire**
## Context

This is part of a wider effort to improve the visibility and documentation of the exports and is a continuation of https://github.com/DFE-Digital/apply-for-teacher-training/pull/4295 and https://github.com/DFE-Digital/apply-for-teacher-training/pull/4333.

We currently have a list of common columns in `_common_columns.yml`. Similar to the issue we saw with the locales strings, this file became very long and it was worth splitting it out. 

## Changes proposed in this pull request

This PR splits out the file into multiple smaller files and updates the logic to load and then merge these files together. Hopefully, this makes the documentation files easier to read and easier to add to in future.

## Guidance to review

There will be some work in a future PR to DRY out the `data_set_documentation` class and to throw exceptions when files aren't found or if shadowed columns are present. I've intentionally kept this PR small and focussed. 

## Link to Trello card

https://trello.com/c/s2pOqGYM/3162-export-documentation

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/services/data_set_documentation.rb
Comment (duncanjbrown) [inline app/services/data_set_documentation.rb:13]: Is it worth pulling this out into a private class method now?

**Review by duncanjbrown — APPROVED** [review]
Great — small refactor if you like!

File: app/services/data_set_documentation.rb
Comment (raamSoftwire) [reply app/services/data_set_documentation.rb:13]: Let's think about that when I do the larger refactor of this class once all the export docs are done.

**Comment by raamSoftwire** [comment]
⚠Please do not merge just yet. I'll merge it in once https://github.com/DFE-Digital/apply-for-teacher-training/pull/4333 is merged

**Pull request merged**
